src/xclim/indices/_hydrology.py

```diff
@@ -685,3 +701,558 @@ def antecedent_precipitation_index(pr: xarray.DataArray, window: int = 7, p_exp:
+    Journal of Open Source Software, 4(39), 1556, https://doi.org/10.21105/joss.01556
+    https://pypi.org/project/pymannkendall/
+
+    Sauquet, E., Evin, G., Siauve, S., Aissat, R., Arnaud, P., Bérel, M., Bonneau, J., Branger, F., Caballero, Y.
+    , Colléoni, F., Ducharne, A., Gailhard, J., Habets, F., Hendrickx, F., Héraut, L., Hingray, B., Huang, P.,
+    Jaouen, T., Jeantet, A., … Vidal, J.-P. (2025). A large transient multi-scenario multi-model ensemble
+    of future streamflow and groundwater projections in France. https://doi.org/10.5194/egusphere-2025-1788.
+    Article in preprint stage.
+    """
+    seasons = ["DJF", "MAM", "JJA", "SON", "Year"]
+
+    def compute_seasonal_stats(x):
+        # Convert to pandas Series with DatetimeIndex
+        x_year = x.resample(time="YS-DEC").mean()
+        x_season = x.resample(time="QS-DEC").mean()
+
+        x_series = x_season.to_series()
+
+        # Create a MultiIndex: year + season (0–3)
+        season_index = (
+            x_series.index.month % 12 // 3  # 0 for DJF, 1 for MAM, etc.
+        )
+        x_df = pd.DataFrame({"value": x_series.values, "season": season_index, "year": x_series.index.year})
+        #  Pivot to shape (n_years, 4 seasons)
+        df_seasons = x_df.pivot(index="season", columns="year", values="value")
+
+        # rename columns
+        df_seasons.index = ["DJF", "MAM", "JJA", "SON"]
+
+        ss_DJF = mk.original_test(df_seasons.iloc[0])
+        ss_MAM = mk.original_test(df_seasons.iloc[1])
+        ss_JJA = mk.original_test(df_seasons.iloc[2])
+        ss_SON = mk.original_test(df_seasons.iloc[3])
+        ss_an = mk.original_test(x_year)
+
+        slopes = [ss_DJF.slope, ss_MAM.slope, ss_JJA.slope, ss_SON.slope, ss_an.slope]
+        pvals = [ss_DJF.p, ss_MAM.p, ss_JJA.p, ss_SON.p, ss_an.p]
+
+        return slopes, pvals
+
+    if qsim is not None:
+        slopes, pvals = compute_seasonal_stats(q)
+        slopes_sim, pvals_sim = compute_seasonal_stats(qsim)
+        slopes_np = np.array(slopes)
+        slopes_sim_np = np.array(slopes_sim)
+        ratio = slopes_np / slopes_sim_np
+        ds = xarray.Dataset(
+            data_vars={
+                "Sen_slope_obs": ("season", slopes),
+                "p_value_obs": ("season", pvals),
+                "Sen_slope_sim": ("season", slopes_sim),
+                "p_value_sim": ("season", pvals_sim),
+                "ratio": ("season", ratio),
+            },
+            coords={"season": seasons},
+        )
+
+    else:
+        slopes, pvals = compute_seasonal_stats(q)
+        # Create labeled xarray
+        ds = xarray.Dataset(
+            data_vars={"Sen_slope": ("season", slopes), "p_value": ("season", pvals)}, coords={"season": seasons}
+        )
+
+    # Assign empty units to all variables
+    for var in ds.data_vars:
+        ds[var].attrs["units"] = ""
+
+    return ds
+
+
+@declare_units(q="[discharge]")
+def annual_maxima(
```

Comment:
In generic.py, there is a `doymax` function that does this. Not sure there is added value in duplicating that functionality, at least within xclim. We could make a different argument for xhydro.